### PR TITLE
fix `Warning: Prop `href` did not match. Server: "#" Client: ...`

### PR DESCRIPTION
**Pull request by B2o5T**

### CODE REVIEW

File: packages/nextra-theme-docs/src/index.tsx [hunk 1]
```diff
@@ -271,12 +269,14 @@ function Layout(props: any): ReactElement {
...
   const extendedConfig = {
     ...DEFAULT_THEME,
-    ...context.themeConfig,
-    unstable_flexsearch: context.pageOpts.unstable_flexsearch,
-    newNextLinkBehavior: context.pageOpts.newNextLinkBehavior
+    ...themeConfig,
+    unstable_flexsearch: pageOpts.unstable_flexsearch,
+    newNextLinkBehavior: pageOpts.newNextLinkBehavior,
+    title: pageOpts.title,
+    meta: pageOpts.meta
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/index.tsx:279]: now title and frontMatter's data (meta) are accessible in theme context

File: packages/nextra-theme-docs/src/types.ts [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
   github?: string
   projectLinkIcon?: ReactNode | FC
   docsRepositoryBase?: string
-  titleSuffix?:
-    | string
-    // Can't be React component, otherwise will get Warning: A title element received an array with more than 1 element as children.
-    | ((props: {
-        config: DocsThemeConfig
-        title: string
-        meta: Meta
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/types.ts:15]: removed these props, since they can be retrieved via `useConfig` hook

Comment (shuding) [reply packages/nextra-theme-docs/src/types.ts:15]: This definitely makes it 100x cleaner!

**Comment by shuding** [comment]
Seems like this PR was wrongly rebased?

**Review by shuding — APPROVED** [review]
Looks great, thanks!

**Pull request merged**